### PR TITLE
Tf 10167/justification form workflow

### PR DESCRIPTION
**Pull request by sramalho94**
### Ticket(s)

_Required unless this is just a change to documentation._

- [10167](https://airtable.com/appfJZShN8K4tcWHU/pagXdnDYi0tVl4u8R?HjEAY=rec9AlbR9XDl3sbdV)

### Type of change

_Please delete any options that are not relevant_
- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation or testing

### Description

This PR aims to increase the utility of our new 3rd party package justification form. Currently, the justification form always appears as part of the PR form, even if there are no new dependencies installed. I have created a workflow that does the following:

  - checks for changes in the `package.json` and `package-lock.json` files in the branch opening the PR
  - If changes are detected, a bot will comment the form on the PR for the dev to fill out

I will revise our documentation once feedback is given, and all the necessary changes to this PR are implemented

### Screenshots

justification_form

#### Please include a screenshot of the Figma design file

No changes to visual design of website, change is purely infrastructural 

### Checklist:

_Please delete any options that are not relevant_

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings in the console
- [X] There are no new linting errors/problems in my code
- [X] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

File: .github/workflows/3rd-party-check.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: 3rd Party Dependency Check
```
Comment (e-papanicolas) [inline .github/workflows/3rd-party-check.yml:1]: Can we add "New" to the beginning of the name.

File: .github/workflows/3rd-party-check.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Post Comment
+        if: ${{ env.DEPENDENCY_CHANGE == 'true' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+            PR_NUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+            echo "PR_NUMBER: $PR_NUMBER"
+  
+            COMMENT_BODY='### New 3rd party dependency justification\n'
+            COMMENT_BODY+='If you are looking to implement a new library as part of your pull request, you **must** fill out the information required in the form. Type your answers between the two `|  |` at the end of each line. _Please delete if not relevant._\n\n'
```
Comment (e-papanicolas) [inline .github/workflows/3rd-party-check.yml:65]: In the bit that says "Please delete if not relevant," let's update this to say something like: "If this is just an update to an already existing package, you can ignore this".

Comment (e-papanicolas) [inline .github/workflows/3rd-party-check.yml:65]: Could we update the initial sentence of this line also, thinking "@PR-author, we have detected new changes to the package.json file, if you are looking...."?

**Review by e-papanicolas — CHANGES_REQUESTED** [review]
@sramalho94 this is really cool and I am so impressed with your dedication to making everything clean and smooth for both contributors and reviewers!

File: .github/workflows/3rd-party-check.yml (hunk 2 above)
Comment (sramalho94) [reply .github/workflows/3rd-party-check.yml]: Implemented changes to workflow
comment

File: .github/workflows/3rd-party-check.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+          # Compare package-lock.json
+          echo "Checking package-lock.json for changes..."
+          if git diff --name-only origin/${{ github.base_ref }}...HEAD | grep 'package-lock.json'; then
+            echo "Changes detected in package-lock.json"
+            echo "DEPENDENCY_CHANGE=true" >> $GITHUB_ENV
+          fi
+
+      - name: Post Comment
+        if: ${{ env.DEPENDENCY_CHANGE == 'true' }}
```
Comment (e-papanicolas) [inline .github/workflows/3rd-party-check.yml:57]: Have you noticed this context access warning for this line?

File: .github/workflows/3rd-party-check.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          PR_AUTHOR="${{ github.event.pull_request.user.login }}"
+          PR_NUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+          echo "PR_NUMBER: $PR_NUMBER"
+  
+          COMMENT_BODY="### New 3rd party dependency justification\n"
+          COMMENT_BODY+="@$PR_AUTHOR, we have detected new changes to the package.json file, if you are looking to implement a new library as part of your pull request, you **must** fill out the information required in the form. Type your answers between the two \`| |\` at the end of each line. _If this is just an update to an already existing package, you can ignore this_\n\n"
```
Comment (e-papanicolas) [inline .github/workflows/3rd-party-check.yml:66]: Can you put a period after "this", thanks!

File: .github/workflows/3rd-party-check.yml (hunk 3 above)
Comment (e-papanicolas) [reply .github/workflows/3rd-party-check.yml:57]: If it is not a problem can ignore.

Comment (sramalho94) [reply .github/workflows/3rd-party-check.yml]: I don't think it's a problem, but having a warning is never a good time. I've resolved it, by setting the conditional if statement to use the outputs from the previous step

**Review by e-papanicolas — APPROVED** [review]

**Pull request merged**